stdlib/public/SDK/Foundation/JSONEncoder.swift

```diff
@@ -10,6 +10,35 @@
 //
 //===----------------------------------------------------------------------===//
 
+/// A marker protocol used to determine whether a value is a `String`-keyed `Dictionary`
+/// containing `Encodable` values (in which case it should be exempt from key conversion strategies).
+#if arch(i386) || arch(arm)
+internal protocol _JSONStringDictionaryEncodableMarker { }
+#else
+fileprivate protocol _JSONStringDictionaryEncodableMarker { }
+#endif
+
+extension Dictionary : _JSONStringDictionaryEncodableMarker where Key == String, Value: Encodable { }
+
+/// A marker protocol used to determine whether a value is a `String`-keyed `Dictionary`
+/// containing `Decodable` values (in which case it should be exempt from key conversion strategies).
+///
+/// The marker protocol also provides access to the type of the `Decodable` values,
+/// which is needed for the implementation of the key conversion strategy exemption.
+#if arch(i386) || arch(arm)
```

Comment:
I'd leave a comment here with a link to the SR you filed regarding this. Also, since this only affected the 32-bit iOS simulator, I think we can cut down on this `#if` a bit in practice, right?

Reply:
In fact, this might be a good candidate for `#if targetEnvironment(simulator)`